File: CSharp/checkoutpage_find_by_merchant_reference_number.cs
// Description: Checkoutpage Find by Merchant Reference Number request sample for Checkout Page API.
var options = new RestClientOptions("https://checkout.bmstaging.id")
{
  MaxTimeout = -1,
};
var client = new RestClient(options);
var request = new RestRequest("/api/findByRef/1257582293", Method.Get);
request.AddHeader("X-Winpay-Timestamp", "2026-11-24T17:07:05+07:00");
request.AddHeader("X-Winpay-Key", "xxxxxxxx");
request.AddHeader("X-Winpay-Signature", "xxxxxxxxxxxxxxxxx");
request.AddHeader("Content-Type", "application/json");
request.AddHeader("Accept", "application/json");
RestResponse response = await client.ExecuteAsync(request);
Console.WriteLine(response.Content);

File: Java/checkoutpage_find_by_merchant_reference_number.java
// Description: Checkoutpage Find by Merchant Reference Number request sample for Checkout Page API.
OkHttpClient client = new OkHttpClient().newBuilder()
  .build();
MediaType mediaType = MediaType.parse("application/json");
RequestBody body = RequestBody.create(mediaType, "");
Request request = new Request.Builder()
  .url("https://sandbox-checkout.winpay.id/api/findByRef/1257582293")
  .method("GET", body)
  .addHeader("X-Winpay-Timestamp", "2026-11-24T17:07:05+07:00")
  .addHeader("X-Winpay-Key", "xxxxxxxx")
  .addHeader("X-Winpay-Signature", "xxxxxxxxxxxxxxxxx")
  .addHeader("Content-Type", "application/json")
  .addHeader("Accept", "application/json")
  .build();
Response response = client.newCall(request).execute();
